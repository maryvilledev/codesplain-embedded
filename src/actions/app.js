export const SET_STATE = 'RESTORE_STATE';

export const setState = (newState) => ({
  type: SET_STATE,
  payload: newState,
});
