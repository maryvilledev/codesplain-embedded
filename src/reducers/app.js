import * as actions from '../actions/app';

export const initialState = {
  annotations: {},
  AST: {},
  filters: {},
  snippetLanguage: 'python3',
  snippet: '',
  snippetKey: '',
  snippetTitle: '',
};

const app = (state = initialState, action) => {
  switch (action.type) {
    case actions.RESTORE_STATE: {
      return Object.assign({}, action.payload);
    }
    default: {
      return state;
    }
  }
}

export default app;
