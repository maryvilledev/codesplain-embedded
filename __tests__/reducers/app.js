import * as actions from '../../src/actions/app';
import reducer, { initialState } from '../../src/reducers/app';

describe(`Reducer: App`, () => {
  describe(`initialState`, () => {
    it(`matches snapshot`, () => {
      expect(initialState).toMatchSnapshot();
    });
  });

  it(`handles SET_STATE`, () => {
    const savedState = {
      snippet: `print('test')`,
      snippetTitle: `Test Snippet`,
      annotations: {},
      AST: {
        type: 'file_input',
        begin: 0,
        end: 1,
        children: [
          'NEWLINE',
          'INDENT',
        ]
      },
      filters: {}
    };
    const action = {
      type: actions.SET_STATE,
      payload: savedState,
    };
    expect(reducer(undefined, action)).toEqual(savedState);
  });

  it(`handles SET_SELECTED_LINE`, () => {
    const selectedLine = 0;
    const expected = {
      ...initialState,
      selectedLine,
    }
    const action = {
      type: actions.SET_SELECTED_LINE,
      payload: selectedLine,
    };
    expect(reducer(undefined, action)).toEqual(expected);
  })
});
