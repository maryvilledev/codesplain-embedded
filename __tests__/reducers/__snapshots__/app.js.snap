// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer: App initialState matches snapshot 1`] = `
Object {
  "AST": Object {},
  "annotations": Object {},
  "filters": Object {},
  "selectedLine": undefined,
  "snippet": "",
  "snippetKey": "",
  "snippetLanguage": "python3",
  "snippetTitle": "",
}
`;
